pkg/engine/operation/resource_node.go

```diff
@@ -23,59 +23,67 @@ var _ ExecutableNode = (*ResourceNode)(nil)
 
 func (rn *ResourceNode) Execute(operation *Operation) status.Status {
 	log.Debugf("execute node:%s", rn.ID)
-	// 1. prepare planedState
-	planedState := rn.state
-	if rn.Action != Delete {
-		if operation.OperationType != Preview {
-			// replace implicit references
-			value := reflect.ValueOf(rn.state.Attributes)
-			_, implicitValue, s := ParseImplicitRef(value, operation.CtxResourceIndex, ImplicitReplaceFun)
-			if status.IsErr(s) {
-				return s
-			}
-			rn.state.Attributes = implicitValue.Interface().(map[string]interface{})
+	if operation.OperationType != DestroyPreview && operation.OperationType != Destroy {
```

Comment:
applyPreview also don't need to replace implicit references according to my knowledge

Reply:
ok...that means only `Apply` need this